src/Symfony/Component/Routing/Annotation/Route.php

```diff
@@ -24,133 +24,59 @@
 #[\Attribute(\Attribute::IS_REPEATABLE | \Attribute::TARGET_CLASS | \Attribute::TARGET_METHOD)]
 class Route
 {
-    private $path;
-    private $localizedPaths = [];
-    private $name;
-    private $requirements = [];
-    private $options = [];
-    private $defaults = [];
-    private $host;
-    private $methods = [];
-    private $schemes = [];
-    private $condition;
-    private $priority;
-    private $env;
+    private ?string $path = null;
+    private array $localizedPaths = [];
+    private array $methods;
+    private array $schemes;
 
     /**
-     * @param array|string      $data         data array managed by the Doctrine Annotations library or the path
-     * @param array|string|null $path
-     * @param string[]          $requirements
-     * @param string[]|string   $methods
-     * @param string[]|string   $schemes
-     *
-     * @throws \BadMethodCallException
+     * @param string[]        $requirements
+     * @param string[]|string $methods
+     * @param string[]|string $schemes
      */
     public function __construct(
-        $data = [],
-        $path = null,
-        string $name = null,
-        array $requirements = [],
-        array $options = [],
-        array $defaults = [],
-        string $host = null,
-        $methods = [],
-        $schemes = [],
-        string $condition = null,
-        int $priority = null,
+        string | array | null $path = null,
```

Comment:
I'd personally prefer no spaces around the `|`. Whatever we decide for this, we need a php-cs-fixer rule for that.

Reply:
I agree with your preference. Currently, PHP CS Fixer (and thus Fabbot) enforces the spaces.

Reply:
Reported at https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5733